rust-lang/cargo
# test: add a regression test for Issue 14409

### What does this PR try to resolve?

Ref https://github.com/rust-lang/cargo/pull/14418#discussion_r1720996541

We should add a regression test for Issue 14409.

### How should we test and review this PR?

Check out the unit test.

If you revert [the fix](https://github.com/rust-lang/cargo/pull/14418), it will fail.

```console
thread 'cargo_info::without_requiring_registry_auth::case' panicked at tests/testsuite/cargo_info/without_requiring_registry_auth/mod.rs:21:10:
Expected success, was 101
stderr:
    Updating `dummy-registry` index
 Downloading crates ...
  Downloaded my-package v0.1.1+my-package (registry `dummy-registry`)
    Updating crates.io index
error: credential provider `false` failed action `get`

Caused by:
  failed to deserialize hello

Caused by:
  EOF while parsing a value at line 1 column 0
```

### Additional information

r? @weihanglo
